### ISSUE TITLE
The handling of duplicate notes

### ISSUE THREAD

**Issue by peavine** [glushchenko/fsnotes#580]
I wondered if it might be desirable to have a uniform approach to dealing with duplicate note names. There are three circumstances where this is an issue.

1) Duplicate an existing note. 

In FSNotes, if you duplicate a note using File > Duplicate, then a date-and-time stamp is added to the note name. For example:

My Note 20190508053952

In this circumstance, the Finder would create:

My Note
My Note Copy
My Note Copy 2

2) A Note already exists. 

In FSNotes, if you move a note to a folder that already contains a note with that name, then a number is added to the moved note name. For example:

My Note
My Note 1
My Note 2

In this circumstance the Finder would prompt and, if "Keep Both" is selected, the Finder would create:

My Note
My Note Copy
My Note Copy 2

3) Create a new note.

In FSNotes, if the user creates multiple new notes with File > New, they all have the note name "Untitled Note" and the note file name is a UUID.

The Finder does not have something directly comparable, but when creating multiple new folders, Finder makes:

untitled folder
untitled folder 2
untitled folder 3

Similarly, when File > New is repeatedly selected in TextEdit, the text documents are named:

Untitled
Untitled 2
Untitled 3

I prefer simple and so, FWIW, my preference would be for FSNotes in circumstances 1 and 2 above to use:

My Note
My Note 1
My Note 2

In circumstance 3 above, for both the note name and note file name, my preference would be to use:

Untitled Note
Untitled Note 1
Untitled Note 2

**Comment by glushchenko** [comment]
Please test https://github.com/glushchenko/fsnotes/releases/tag/5beta1

**Comment by peavine** [comment]
I've done some preliminary testing of the changes referenced by this thread and everything seems to work well. I did encounter two possible cosmetic issues, which can be reproduced as follows:

1) Create Test Folder 1 and Test Folder 2.
2) In Test Folder 1 create a new note named Test Note.
3) With the mouse, drag Test Note from Test Folder 1 to Test Folder 2.

After doing the above Test Note is still shown in Test Folder 1 even though it has been moved to Test Folder 2. Also, when dragging the note to Test Folder 2, the cursor contains a "+" which would seem to indicate the note is being copied rather than moved.

BTW, a lot of awesome new features in FSNotes 5 :)
